scripts/generate-jsonschema.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+set -eu
+
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+GO111MODULE=on go get github.com/chrusty/protoc-gen-jsonschema/cmd/protoc-gen-jsonschema
+
+json_schema_dir=../jsonschema
+autogen_dir="$(bazel info output_base)"/external/deploymentmanager-autogen
+
+protoc --jsonschema_out=json_fieldnames,disallow_additional_properties:$json_schema_dir --proto_path=$autogen_dir --proto_path=$PWD resource.proto
+
+# Fixes the references to definitions in the json schema and restricts some
+# fields to enum values
```

Comment:
1. What happens if we miss a "fix" below? Do we just fail to catch that kind of mistake in autogen specs, or do we fail to validate a valid autogen spec using it?
2. How did we identify these enums? Is there a jq "paths(scalars)" command or something we can reference here to help assist with future updates to this section?

Reply:
1. Clarified this, we are fixing the way the jsonschema gen tool creates the "$ref" keys. This should always work unless there are backwards incompatible change to the jsonschema gen tool.
2. These enums are what we currently support in the mpdev code. Unfortunately we'd have to manually update these enums, but this is better than allowing the user to set invalid values.